ringtail/storagemanager.py

```diff
@@ -3153,46 +3127,49 @@ def _generate_ligand_filtering_query(self, ligand_filters: dict) -> str:
             str: SQLite-formatted query
         """
 
-        sql_ligand_string = "SELECT L.LigName FROM Ligands L WHERE"
+        sql_ligand_string = "SELECT L.LigName FROM Ligands L WHERE "
+        query_list = []
         if "ligand_operator" in ligand_filters:
-            logical_operator = ligand_filters["ligand_operator"]
+            if ligand_filters["ligand_operator"] in ["OR", "AND"]:
+                logical_operator = ligand_filters["ligand_operator"]
         else:
             self.logger.info(
                 "A logical operator to combine ligand filters were not provided, will use the default value 'OR'."
             )
             logical_operator = "OR"
-        if logical_operator is None:
-            logical_operator = "AND"
-        for kw in ligand_filters.keys():
-            fils = ligand_filters[kw]
-            if kw == "ligand_name":
-                for name in fils:
-                    if name == "":
-                        continue
-                    name_sql_str = " L.LigName LIKE '%{value}%' OR".format(value=name)
-                    sql_ligand_string += name_sql_str
-            if kw == "ligand_max_atoms" and ligand_filters[kw] is not None:
-                maxatom_sql_str = " mol_num_hvyatms(ligand_rdmol) <= {} {}".format(
-                    ligand_filters[kw], logical_operator
-                )
-                sql_ligand_string += maxatom_sql_str
-            if kw == "ligand_substruct":
-                for smarts in fils:
+
+        for keyword, filter in ligand_filters.items():
+            if keyword == "ligand_name":
+                # make each name a partial sql string in list format
+                names = [
+                    f"L.LigName LIKE '%{name}%'" for name in filter if filter is not ""
+                ]
+                query_list.append(" OR ".join(names))
```

Comment:
If there are multiple name filters, all of them need to be embedded in Parentheses. 
```py
                query_list.append(f"( {" OR ".join(names)} ) ")
```